native/NativeSampleApps/test/RestExplorerTest/src/com/salesforce/samples/restexplorer/ExplorerActivityTest.java

```diff
@@ -120,32 +120,60 @@ public void setUp() throws Exception {
         eq = new EventsListenerQueue();
 
         // Waits for app initialization to complete.
-        if (SalesforceSDKManager.getInstance() == null) {
+        if (!SalesforceSDKManager.hasInstance()) {
             eq.waitForEvent(EventType.AppCreateComplete, 5000);
         }
         targetContext = getInstrumentation().getTargetContext();
         clientManager = new ClientManager(targetContext, targetContext.getString(R.string.account_type), null, SalesforceSDKManager.getInstance().shouldLogoutWhenTokenRevoked());
         clientManager.createNewAccount(TEST_ACCOUNT_NAME, TEST_USERNAME, TEST_REFRESH_TOKEN,
                 TEST_ACCESS_TOKEN, TEST_INSTANCE_URL, TEST_LOGIN_URL, TEST_IDENTITY_URL, TEST_CLIENT_ID, TEST_ORG_ID, TEST_USER_ID, null);
-        mockHttpAccessor = new MockHttpAccess(SalesforceSDKManager.getInstance().getAppContext());
         SalesforceSDKManager.getInstance().getPasscodeManager().setTimeoutMs(0 /* disabled */);
 
-        // Plug our mock HTTP accessor.
+        // Plug a modified okHttpClient that doesn't actually go to the server.
         final ExplorerActivity activity = getActivity();
         assertNotNull("Activity should not be null", activity);
         final RestClient client = activity.getClient();
         assertNotNull("Rest client should not be null", client);
-        client.setHttpAccessor(mockHttpAccessor);
+        final OkHttpClient mockOkHttpClient = buildMockOkHttpClient();
+        client.setOkHttpClient(mockOkHttpClient);
+    }
+
+    private OkHttpClient buildMockOkHttpClient() {
```

Comment:
Instead of a mock http accessor. We use a "mock" okhttpclient.
Actually it's a real okHttpClient that has an interceptor which never sends the request to the server but instead builds a response using the request body.